### ISSUE TITLE
Support for unflatten?

### ISSUE THREAD

**Issue by Fetix** [xflr6/graphviz#122]
Hi there,

love your package to draw genealogy graphs.
Sometimes they happen to be quite wide and so I came to wonder if there a way to use GraphViz's unflatten via graphviz.

Thanks a lot for any suggestions and help.
Best,
Fetix

**Comment by xflr6** [comment]
Thanks Fetix, glad if this is useful.

As [unflatten](https://manpages.debian.org/testing/graphviz/unflatten.1.en.html) both reads and writes DOT  syntax, I think we can at least do something along the following lines (using an [example from a related SO question](https://stackoverflow.com/questions/11134946)):

```python
In [1]: import graphviz
   ...: 
   ...: g = graphviz.Digraph(graph_attr={'nodesep': '0.3',
   ...:                                  'ranksep': '0.2',
   ...:                                  'margin': '0.1'},
   ...:                      node_attr={'shape': 'rectangle'},
   ...:                      edge_attr={'arrowsize': '0.8'})
   ...: 
   ...: g.edges(('1', str(i)) for i in range(2, 11))
   ...: 
   ...: print(g.source)
   ...: g
digraph {
	graph [margin=0.1 nodesep=0.3 ranksep=0.2]
	node [shape=rectangle]
	edge [arrowsize=0.8]
	1 -> 2
	1 -> 3
	1 -> 4
	1 -> 5
	1 -> 6
	1 -> 7
	1 -> 8
	1 -> 9
	1 -> 10
}
Out[1]: 
```
![g](https://user-images.githubusercontent.com/6342379/102470829-bdc27380-4054-11eb-8fbf-b5966d4e6c77.png)

```python
In [2]: def unflatten(source, encoding='utf-8'):
   ...:     cmd = ['unflatten', '-l', '3']
   ...:     out, err = graphviz.backend.run(cmd, input=source,
   ...:                                     capture_output=True,
   ...:                                     encoding=encoding)
   ...:     return graphviz.Source(out)
   ...: 
   ...: s = unflatten(g.source)
   ...: 
   ...: print(s.source)
   ...: s
digraph {
	graph [margin=0.1,
		nodesep=0.3,
		ranksep=0.2
	];
	node [shape=rectangle];
	edge [arrowsize=0.8];
	1 -> 2	 [minlen=1];
	1 -> 3	 [minlen=2];
	1 -> 4	 [minlen=3];
	1 -> 5	 [minlen=1];
	1 -> 6	 [minlen=2];
	1 -> 7	 [minlen=3];
	1 -> 8	 [minlen=1];
	1 -> 9	 [minlen=2];
	1 -> 10	 [minlen=3];
}

Out[2]: 
``` 

![s](https://user-images.githubusercontent.com/6342379/102471136-1c87ed00-4055-11eb-9f3e-193ddfde871d.png)

Marking this as enhancement.  Not sure if this might be a `Graph`-method (cf. [PyGraphviz](https://pygraphviz.github.io/documentation/pygraphviz-1.6/reference/agraph.html#pygraphviz.AGraph.unflatten)) or rather one on `Source`.

Note that this library does not parse DOT syntax, so the processed result cannot support modification via with graph-editing methods (`node()`, `edge()`, etc).

**Comment by Fetix** [comment]
Hi xflr6,
thanks for your quick reply and time taken, I really appreciate it.
That's awesome. Thanks for providing a solution along with a working example, love it.

**Comment by xflr6** [comment]
Thanks. Re-opened (in my understanding this is about adding e.g. an `unflatten()`-method to this library: the code above is just a proof of concept).

Can you give some advice on the arguments? E.g. does it makes sense to run `unflatten` without an argument (IIUC that only round-trips the DOT but does not change something semantically). Any idea for the argument names corresponding to `-l len`, `-f`, and `-c len`? Maybe you would like to contribute here?

**Comment by xflr6** [comment]
Added `.unflatten()` to `Graph`, `Digraph`, and `Source`: f31715796d509382ba33880ddbbe79f1a888310a

**Comment by xflr6** [comment]
Resolved in [0.16 release](https://pypi.org/project/graphviz/0.16/#files) (54b4eecd0f2cc2ac356943ecc1f042abb6e627d2).
